### ISSUE TITLE
Periodically with specific date and time

### ISSUE THREAD

**Issue by rrifafauzikomara** [fluttercommunity/flutter_workmanager#207]
I see on the example have periodically task, like this.

```
//This task runs periodically
              //It will run about every hour
              PlatformEnabledButton(
                  platform: _Platform.android,
                  child: Text("Register 1 hour Periodic Task"),
                  onPressed: () {
                    Workmanager.registerPeriodicTask(
                      "5",
                      simplePeriodic1HourTask,
                      frequency: Duration(hours: 1),
                    );
                  }),
```

On the above function, we know that will be run the task periodically every hour. And the question is, can we use this plugin to run the task scheduled? For example, I want to run the task every 9 PM.

**Comment by gebsl** [comment]
Yes and no. It is possible, but not with `registerPeriodicTask`. Android's WorkManager is not built to run tasks in a periodic manner. But you can achieve something similar, if you register a task via `registerOneOffTask`. Inside the callback you schedule another one off task, calaculating the timespan for your next execution.

I think you should go for Android's [AlarmManager](https://developer.android.com/reference/android/app/AlarmManager), as it is designed to run code at a specific date and time.

Anyway, if you really want to use WorkManager package, there is some code I came up with below.
**Important**: Please note I haven't tested or tried this code. I don't know if this will work, it's only an idea.

```dart
void callbackDispatcher() {
  Workmanager.executeTask((task, inputData) {
    // calculate new delay here
    // e.g. there is only 8 hours left until 9 PM, so you pass a duration of 8 hours
    // the next time callbackDispatcher is called, the initial delay might be different, as Android does not guarantee that your task will be run at a specific time.
    Workmanager.registerOneOffTask("1", "simpleTask", initialDelay: Duration(hours: 8));
    return Future.value(true);
  });
}

void main() {
  Workmanager.initialize(
    callbackDispatcher,
  );
  Workmanager.registerOneOffTask("1", "simpleTask", initialDelay: Duration(hours: 12));
  runApp(MyApp());
}
```

And please, if you want to create github issues: Give them meaningful titles, as otherwise other users won't be able to figure out what the issue is all about, thanks.

**Comment by rrifafauzikomara** [comment]
@gebsl Thanks for the answer, but If using `initialDelay` I can't. Because I need to specifically run the task `every 9 PM`.

**Comment by gebsl** [comment]
Yeah, that's what I wanted to say. You always have to calculate the initial delay according to when you want to run the task. But anyway, Android will not guarantee the task will really run at this time. It can also decide to schedule it another 2 hours later.

That's why you should look at Android's AlarmManager, this is exactly what you need.

**Comment by rrifafauzikomara** [comment]
I know how to do that on Android native, but the real case I need to run the task on both Android on iOS. So this package support only for Android right? Because on the example all tasks have label "Android Only", it's mean on the iOS for now not supported?

*Still to try to find the package first, if it can't, I will create using method channel.

**Comment by gebsl** [comment]
Exaclty, iOS is rather limited that's why most of the functions are only available on Android.

**Comment by rrifafauzikomara** [comment]
Okay, Thanks @gebsl for discussion about this package!

**Comment by rrifafauzikomara** [comment]
> Yes and no. It is possible, but not with `registerPeriodicTask`. Android's WorkManager is not built to run tasks in a periodic manner. But you can achieve something similar, if you register a task via `registerOneOffTask`. Inside the callback you schedule another one off task, calaculating the timespan for your next execution.
> 
> I think you should go for Android's [AlarmManager](https://developer.android.com/reference/android/app/AlarmManager), as it is designed to run code at a specific date and time.
> 
> Anyway, if you really want to use WorkManager package, there is some code I came up with below.
> **Important**: Please note I haven't tested or tried this code. I don't know if this will work, it's only an idea.
> 
> ```dart
> void callbackDispatcher() {
>   Workmanager.executeTask((task, inputData) {
>     // calculate new delay here
>     // e.g. there is only 8 hours left until 9 PM, so you pass a duration of 8 hours
>     // the next time callbackDispatcher is called, the initial delay might be different, as Android does not guarantee that your task will be run at a specific time.
>     Workmanager.registerOneOffTask("1", "simpleTask", initialDelay: Duration(hours: 8));
>     return Future.value(true);
>   });
> }
> 
> void main() {
>   Workmanager.initialize(
>     callbackDispatcher,
>   );
>   Workmanager.registerOneOffTask("1", "simpleTask", initialDelay: Duration(hours: 12));
>   runApp(MyApp());
> }
> ```
> 
> And please, if you want to create github issues: Give them meaningful titles, as otherwise other users won't be able to figure out what the issue is all about, thanks.

Hi @gebsl 

I check the example for `registerOneOffTask` method, and then I clone the example. But the `registerOneOffTask` method can't run on iOS (the button disable), it's mean this package only support Android platform?

[![enter image description here][1]][1]


  [1]: https://i.stack.imgur.com/gtFAM.png

**Comment by gebsl** [comment]
Please read the documentation carefully -> https://github.com/fluttercommunity/flutter_workmanager/blob/master/IOS_SETUP.md#enabling-background-fetch

Also: https://github.com/fluttercommunity/flutter_workmanager/issues/109

To conclude: No, oneOff tasks are **currently** not supported on iOS. This might change, according to #109.

**Comment by rrifafauzikomara** [comment]
Thanks @gebsl 

I have the answer.

> **One off tasks** or **Periodic tasks** are available on Android only for now! (see #109)
